**1) DATA DESCRIPTION**

In [1]:
library(tidyverse)
# players dataset description
players <- read_csv("players.csv")
players_rows <- nrow(players)
players_rows
experience_summary <- table(players$experience)
subscribe_summary <- table(players$subscribe)
played_hours_mean <- mean(players$played_hours)
gender_summary <- table(players$gender)
Age_mean <- mean(players$Age, na.rm = TRUE)
experience_summary
subscribe_summary
played_hours_mean
gender_summary
Age_mean

# sessions dataset description
sessions <- read_csv("sessions.csv")
sessions_rows <- nrow(sessions)
sessions_rows
original_start_time_mean <- mean(sessions$original_start_time, na.rm = TRUE)
original_end_time_mean <- mean(sessions$original_end_time, na.rm = TRUE)
original_start_time_mean
original_end_time_mean

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 196


 Amateur Beginner      Pro  Regular  Veteran 
      63       35       14       36       48 


FALSE  TRUE 
   52   144 

[1] 5.845918


          Agender            Female              Male        Non-binary 
                2                37               124                15 
            Other Prefer not to say      Two-Spirited 
                1                11                 6 

[1] 21.13918

Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


[1] 1535

[1] 1.719201e+12

[1] 1.719196e+12

The dataset consists of two files, the players.csv file which includes data on each individual player and the sessions.csv file which includes the unique play sessions of each individual player as well as data about the session. I have organized the data description by file below. 

The **players.csv** file consists of 196 observations and has the following variables:
|Variable Name|Variable Type|Variable Meaning|
|-------------|-------------|----------------|
|experience|chr|indicates the experience level of the individual player described using 5 categories: Amateur, Beginner, Pro, Regular and Veteran|
|subscribed|lgl|This variable indicates whether or not the individual player is subscribed or not|
|hashedEmail|chr|This is the individual's hashed email|
|played_hours|dbl|This variable reports the amount of hours played on MineCraft by each individual|
|name|chr|This reports each inidivual player's name|
|gender|chr|This reports each individual player's gender|
|Age|dbl|This reports each individual player's age|

And the following summary statistics:
|Variable|Summary|
|--------|----|
|experience|In summary, this dataset has 63 amateur players, 35 beginners, 14 pros, 36 regulars and 48 veteran players|
|subscribed|52 players were unsubscribed and 144 were subscribed|
|hashedEmail|N/A|
|played_hours|the average number of hours played amongst all the individuals was 5.85 hours|
|name|N/A|
|gender|Amongst the players 37 were female, 124 male, 2 agender, 15 non-binary, 6 two-spirited, 11 preferred not to say and 1 other|
|Age|the average age of the players is 21.14|

The **sessions.csv** file consists of 1535 observations and has the following variables:
|Variable Name|Variable Type|Variable Meaning|
|-------------|-------------|----------------|
|hashedEmail|chr|This reports the hashed email for each unique player|
|start_time|chr|This indicates the start time and day for each individual player's session|
|end_time|chr|This indicates the end time and day for each individual player's session|
|original_start_time|dbl|This variable indicates the same values as in the start_time but in milliseconds|
|original_end_time|dbl|This variable indicates the same values as in the end_time but in milliseconds|

and the following summary statistics:
|Variable|Summary|
|--------|-------|
|hashedEmail|N/A|
|start_time|N/A|
|end_time|N/A|
|original_start_time|the average start time in UNIX time (milliseconds) is 1.72E+12|
|original_end_time|the average end time in UNIX time (milliseconds) is 1.72E+12|

Some issues that I see in the data, that could potential become issues in the future is the UNIX time values for the sessions file part of the dataset, this is because since the numbers are soo big, it's hard to see that there's any difference between values if you're rounding the numbers to 2 decimal places.

All this data was collected by creating a MineCraft server that tracks and record the individual player's actions as they play MineCraft in this specific server. 

**2) QUESTIONS**

The one broad question I will be addressing is question 3: *What time windows are most likely to have large numbers of simultaneous players?* The specific question that I have formulated based on this broad question is: *Can the experience level of a player predict the peak in simultaneous players in the MineCraft datasets?* The data will help me address my question as I can compare the experience levels of individual players to the time of day that they log into the server and I can use these variables to see if there is an overlap between the experience level of the player and their session time. This can overlap can help determine if there is a correlation between experience level peak time that simultaneous players were active. 